pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -228,4 +240,20 @@ public String getTableName() {
   public File getTableDataDir() {
     return _indexDir;
   }
+
+  @Override
+  public void addSegmentError(String segmentName, SegmentErrorInfo segmentErrorInfo) {
+    _errorCache.put(new Pair<>(_tableNameWithType, segmentName), segmentErrorInfo);
+  }
+
+  @Override
+  public Map<String, SegmentErrorInfo> getSegmentErrors() {
+    if (_errorCache == null) {
+      return Collections.emptyMap();
+    } else {
+      // Filter out entries that match the table name.
+      return _errorCache.asMap().entrySet().stream().filter(map -> map.getKey().getFirst().equals(_tableNameWithType))
```

Comment:
Avoid `stream` apis for performance concern

Reply:
In general I do, I took the liberty here because it is not in critical path of any performance tasks.